lib/ansible/playbook/taggable.py

```diff
@@ -32,21 +30,6 @@ class Taggable:
     untagged = frozenset(['untagged'])
     _tags = FieldAttribute(isa='list', default=list, listof=(string_types, int), extend=True)
 
-    def __init__(self):
-        super(Taggable, self).__init__()
-
-    def _load_tags(self, attr, ds):
-        if isinstance(ds, list):
-            return ds
-        elif isinstance(ds, string_types):
-            value = ds.split(',')
```

Comment:
comma-separated lists won't work anymore

Reply:
That is handled already in `evaluate_tags`.  It's just not visible in this PR since I didn't change it.

Reply:
Tags is a list though. But it's never iterated over with the string types being separated by ','. I don't actually see where that `if not isinstance(tags, list):` in evaluate_tags would be true.

Reply:
Simple test case:
```
---
- hosts: localhost
  connection: local
  gather_facts: no
  tasks:
  - debug:
        var: ansible_host
    tags: one,two,three
```
Works on devel, but not this branch.

Reply:
Due to where `tags` is used, `post_validate` doesn't ensure it's a list before use.  `evaluate_tags` happens before that.  We were doing it in `_load_tags` which happened before `post_validate` too.

As of now it works to do `tags: foo,bar`.  We could go through some deprecation period to disallow using a CSV, but not really related to this PR.

Reply:
Devel:
```
(python2.7.13) 11:15:27 [ansible]$ ansible-playbook test_tag.yml --tags one
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match
'all'


PLAY [localhost] *******************************************************************************************************

TASK [debug] ***********************************************************************************************************
ok: [localhost] => {
    "ansible_host": "127.0.0.1"
}

PLAY RECAP *************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0
```
This branch:
```
(python2.7.13) 11:16:07 [ansible]$ ansible-playbook test_tag.yml --tags one
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match
'all'


PLAY [localhost] *******************************************************************************************************

PLAY RECAP *************************************************************************************************************

```

Reply:
I think this helps.  I was misunderstanding what you were saying.  I must have overlooked something.  I'll see about deduping code but adding `_load_tags` back.

Reply:
we want to keep `tags: x,y` we allow 'string comma lists' everywhere we allow lists, not sure why we want to make exception here